sm/generate_data.py

```diff
@@ -0,0 +1,126 @@
+from pprint import pprint
```

Comment:
Thanks for including this in this PR, I lost it through splitting and combining other PRs. Could we put this in maybe `test/data/` with a small README?

Reply:
Sure!

Reply:
Done!